crates/ruma-common/src/events/relation.rs

```diff
@@ -6,45 +6,51 @@ use js_int::UInt;
 use serde::{Deserialize, Serialize};
 
 use super::AnySyncMessageLikeEvent;
-use crate::{serde::Raw, MilliSecondsSinceUnixEpoch, OwnedEventId, OwnedUserId};
+use crate::{
+    serde::{Raw, StringEnum},
+    MilliSecondsSinceUnixEpoch, OwnedEventId, OwnedUserId, PrivOwnedStr,
+};
 
-/// Summary of all reactions with the given key to an event.
-#[derive(Clone, Debug, Default, Deserialize, PartialEq, Eq, Serialize)]
+/// Summary of all annotations to an event with the given key and type.
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
 #[cfg(feature = "unstable-msc2677")]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-pub struct BundledReaction {
-    /// The key used for the reaction.
+pub struct BundledAnnotation {
+    /// The key used for the annotation.
     pub key: String,
 
-    /// Time of the bundled reaction being compiled on the server.
+    /// The type of the annotation.
+    #[serde(rename = "type")]
+    pub annotation_type: AnnotationType,
+
+    /// Time of the bundled annotation being compiled on the server.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub origin_server_ts: Option<MilliSecondsSinceUnixEpoch>,
 
-    /// Number of reactions.
+    /// Number of annotations.
     pub count: UInt,
 }
 
 #[cfg(feature = "unstable-msc2677")]
-impl BundledReaction {
-    /// Creates a new `BundledReaction`.
-    pub fn new(
-        key: String,
-        origin_server_ts: Option<MilliSecondsSinceUnixEpoch>,
-        count: UInt,
-    ) -> Self {
-        Self { key, origin_server_ts, count }
+impl BundledAnnotation {
+    /// Creates a new `BundledAnnotation` for a reaction.
+    pub fn reaction(key: String, count: UInt) -> Self {
```

Comment:
We should also have a `fn` new with type, key, count parameters (and implement this one in terms of that one).